lib/i18n/backend/base.rb

```diff
@@ -59,18 +59,7 @@ def localize(locale, object, format = :default, options = {})
           format  = I18n.t(:"#{type}.formats.#{key}", options)
         end
 
-        # format = resolve(locale, object, format, options)
-        format = format.to_s.gsub(/%[aAbBpP]/) do |match|
-          case match
-          when '%a' then I18n.t(:"date.abbr_day_names",                  :locale => locale, :format => format)[object.wday]
-          when '%A' then I18n.t(:"date.day_names",                       :locale => locale, :format => format)[object.wday]
-          when '%b' then I18n.t(:"date.abbr_month_names",                :locale => locale, :format => format)[object.mon]
-          when '%B' then I18n.t(:"date.month_names",                     :locale => locale, :format => format)[object.mon]
-          when '%p' then I18n.t(:"time.#{object.hour < 12 ? :am : :pm}", :locale => locale, :format => format).upcase if object.respond_to? :hour
-          when '%P' then I18n.t(:"time.#{object.hour < 12 ? :am : :pm}", :locale => locale, :format => format).downcase if object.respond_to? :hour
-          end
-        end
-
+        format = translate_format(locale, format, object)
```

Comment:
Do you mean that it would be better to have access to options inside translate_format for future customisation? `format` does need default value it will always set from parent method